packages/contact-center/cc-components/src/components/UserState/user-state.tsx

```diff
@@ -7,33 +7,94 @@ import './user-state.scss';
 import {SelectNext, Text} from '@momentum-ui/react-collaboration';
 import {Item} from '@react-stately/collections';
 import {Icon} from '@momentum-design/components/dist/react';
+
 const UserStateComponent: React.FunctionComponent<IUserState> = (props) => {
-  const {idleCodes, setAgentStatus, isSettingAgentStatus, elapsedTime, lastIdleStateChangeElapsedTime, currentState} =
-    props;
+  const {
+    idleCodes,
+    setAgentStatus,
+    isSettingAgentStatus,
+    elapsedTime,
+    lastIdleStateChangeElapsedTime,
+    currentState,
+    customState,
+  } = props;
+
+  let selectedKey;
+  if (customState) {
+    selectedKey = `hide-${customState.developerName}`;
+  } else {
+    selectedKey = currentState;
+  }
+
+  const items = customState
+    ? [{name: customState.name, id: `hide-${customState.developerName}`, developerName: customState.developerName}]
+    : [];
+
+  for (const item of idleCodes) {
+    if (item.name === 'RONA' && item.id === currentState) {
+      selectedKey = `hide-${item.id}`;
```

Comment:
This is the only way we could show RONA and custom options selected while keeping it hidden from the list. It is hacky but had no other choice